src/app/route.ts

```diff
@@ -1,94 +1,104 @@
-  const walletAddress = searchParams.get("walletAddress")
-  const siloEngineAccountId = searchParams.get("silo")
+  const apiKey = searchParams.get("apiKey");
+  const walletAddress = searchParams.get("walletAddress");
+  const siloEngineAccountId = searchParams.get("silo");
 
   // Optional query parameters
-  const fiatAmount = searchParams.get("fiatAmount")
-  const fiatCurrency = searchParams.get("fiatCurrency")
-  const defaultFiatAmount = searchParams.get("defaultFiatAmount")
-  const defaultCryptoAmount = searchParams.get("defaultCryptoAmount")
-  const environment = searchParams.get("environment")
+  const fiatAmount = searchParams.get("fiatAmount");
+  const fiatCurrency = searchParams.get("fiatCurrency");
+  const defaultFiatAmount = searchParams.get("defaultFiatAmount");
+  const defaultCryptoAmount = searchParams.get("defaultCryptoAmount");
+  const environment = searchParams.get("environment");
+  const referrerDomain = req.headers.get("referer")?.split("/")[2];
 
   if (!apiKey) {
-    return missingParameterResponse("apiKey")
+    return missingParameterResponse("apiKey");
+  }
+  if (!referrerDomain) {
+    return new NextResponse("The `referrerDomain` header is required", {
+      status: 400,
+    });
   }
 
   if (!walletAddress) {
-    return missingParameterResponse("walletAddress")
+    return missingParameterResponse("walletAddress");
   }
 
   if (!siloEngineAccountId) {
-    return missingParameterResponse("silo")
+    return missingParameterResponse("silo");
   }
 
-  const targetAddress = getAddress(walletAddress)
+  const targetAddress = getAddress(walletAddress);
 
-  const url = new URL("https://global.transak.com/")
+  const url =
+    environment === "production"
+      ? new URL("https://api-gateway.transak.com/")
+      : new URL("https://api-gateway-stg.transak.com/");
 
-  url.searchParams.set("network", "near")
-  url.searchParams.set("apiKey", apiKey)
-  url.searchParams.set("disableWalletAddressForm", "true")
+  url.searchParams.set("network", "near");
+  url.searchParams.set("apiKey", apiKey);
+  url.searchParams.set("referrerDomain", referrerDomain);
+  url.searchParams.set("disableWalletAddressForm", "true");
   url.searchParams.set(
     "walletAddress",
-    await getForwarderAddress(targetAddress, siloEngineAccountId),
-  )
+    await getForwarderAddress(targetAddress, siloEngineAccountId)
+  );
 
   if (fiatAmount) {
-    url.searchParams.set("fiatAmount", fiatAmount)
+    url.searchParams.set("fiatAmount", fiatAmount);
   }
 
   if (fiatCurrency) {
-    url.searchParams.set("fiatCurrency", fiatCurrency)
+    url.searchParams.set("fiatCurrency", fiatCurrency);
   }
 
   if (defaultFiatAmount) {
-    url.searchParams.set("defaultFiatAmount", defaultFiatAmount)
+    url.searchParams.set("defaultFiatAmount", defaultFiatAmount);
   }
 
   if (defaultCryptoAmount) {
-    url.searchParams.set("defaultCryptoAmount", defaultCryptoAmount)
+    url.searchParams.set("defaultCryptoAmount", defaultCryptoAmount);
```

Comment:
I've mentioned this more but I think you need to look at your prettier plugin. This PR now contains a load of irrelevant changes with these semi colons on every line!